src/value/value.rs

```diff
@@ -394,6 +426,35 @@ impl Value {
         }
     }
 
+    pub(crate) fn is_none(&self) -> bool {
+        matches!(self, Value::Empty(_, Empty::None))
+    }
+
+    /// Attempts to retrieve a nested value through dictionary key or array index.
+    pub(crate) fn index(self, key: &str) -> Option<Value> {
+        fn try_remove<T>(mut vec: Vec<T>, key: &str) -> Option<T> {
+            let index = key.parse().ok()?;
+            vec.get(index)?;
+
+            Some(vec.swap_remove(index))
```

Comment:
This hides a bit of the logic. I'd prefer something like:

```suggestion
            match vec.get(index) {
                Some(_) => Some(vec.swap_remove(index)),
                None => None,
            }
```

Unfortunately the `take()` methods are unstable.